pkg/operator/cluster/controller.go

```diff
@@ -180,7 +182,62 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 }
 
 func (c *ClusterController) onDelete(obj interface{}) {
-	logger.Infof("deleting a cluster not implemented")
+	clusterOrig := obj.(*Cluster)
+
+	// NEVER modify objects from the store. It's a read-only, local cache.
+	// Use scheme.Copy() to make a deep copy of original object.
+	copyObj, err := c.scheme.Copy(clusterOrig)
+	if err != nil {
+		logger.Errorf("failed to create a deep copy of cluster object: %v\n", err)
+		return
+	}
+	cluster := copyObj.(*Cluster)
+
+	logger.Infof("cluster in namespace %s is being deleted.  Waiting for agents to perform cleanup operations.", cluster.Namespace)
+
+	operatorNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	retryCount := 0
+	maxRetry := 15
+	sleepTime := 2
+	for {
+		// TODO: filter this List operation by cluster namespace on the server side
+		vols, err := c.volumeAttachmentController.List(operatorNamespace)
+		if err != nil {
+			logger.Errorf("failed to get volume attachments for operator namespace %s: %+v", operatorNamespace, err)
+			break
+		}
+
+		// find volume attachments in the deleted cluster
+		attachmentsExist := false
+	AttachmentLoop:
```

Comment:
ohh i didnt know you could name loops. Cool!

Reply:
yep, this is the idiomatic way to break out of an outer loop from a nested loop.